service/storage_proxy.cc

```diff
@@ -5690,6 +5690,8 @@ storage_proxy::query_partition_key_range_concurrent(storage_proxy::clock_type::t
 
             exec.push_back(::make_shared<never_speculating_read_executor>(schema, cf.shared_from_this(), p, erm, cmd, std::move(range), cl, std::move(filtered_endpoints), trace_state, permit, std::monostate()));
             ranges_per_exec.emplace(exec.back().get(), std::move(merged_ranges));
+
+            co_await coroutine::maybe_yield();
```

Comment:
I don't know if this is enough to fix the stall.  Note there is another loop on `i` above, so if we spend most of our iteration time there, we won't hit the maybe_yield here.

With 3 nodes it's likely everything was contiguous with RF=3.

Reply:
good point, I'll add one in the inner loop (that is execute only if the replication strategy does not `uses_tablets`).

Reply:
fixed in v3